extensions/common/sts-account-provisioner/src/main/java/org/eclipse/edc/identityhub/common/provisioner/StsAccountProvisionerExtension.java

```diff
@@ -0,0 +1,114 @@
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Metaform Systems, Inc. - initial API and implementation
+ *
+ */
+
+package org.eclipse.edc.identityhub.common.provisioner;
+
+import org.eclipse.edc.iam.identitytrust.sts.spi.store.StsClientStore;
+import org.eclipse.edc.identithub.spi.did.DidDocumentService;
+import org.eclipse.edc.identityhub.spi.keypair.KeyPairService;
+import org.eclipse.edc.identityhub.spi.keypair.events.KeyPairRevoked;
+import org.eclipse.edc.identityhub.spi.keypair.events.KeyPairRotated;
+import org.eclipse.edc.identityhub.spi.participantcontext.AccountProvisioner;
+import org.eclipse.edc.identityhub.spi.participantcontext.events.ParticipantContextDeleted;
+import org.eclipse.edc.runtime.metamodel.annotation.Extension;
+import org.eclipse.edc.runtime.metamodel.annotation.Inject;
+import org.eclipse.edc.runtime.metamodel.annotation.Provider;
+import org.eclipse.edc.spi.event.EventRouter;
+import org.eclipse.edc.spi.security.Vault;
+import org.eclipse.edc.spi.system.ServiceExtension;
+import org.eclipse.edc.spi.system.ServiceExtensionContext;
+import org.eclipse.edc.transaction.spi.TransactionContext;
+import org.jetbrains.annotations.Nullable;
+
+import java.security.SecureRandom;
+
+import static java.util.Optional.ofNullable;
+import static org.eclipse.edc.identityhub.common.provisioner.StsAccountProvisionerExtension.NAME;
+
+@Extension(value = NAME)
+public class StsAccountProvisionerExtension implements ServiceExtension {
+    public static final String NAME = "STS Account Provisioner Extension";
+    public static final int DEFAULT_CLIENT_SECRET_LENGTH = 16;
+    @Inject
+    private EventRouter eventRouter;
+    @Inject
+    private KeyPairService keyPairService;
+    @Inject
+    private DidDocumentService didDocumentService;
+    @Inject(required = false)
+    private StsClientStore stsClientStore;
+    @Inject
+    private Vault vault;
+    @Inject(required = false)
+    private StsClientSecretGenerator stsClientSecretGenerator;
+    @Inject
+    private TransactionContext transactionContext;
+
+    private StsAccountProvisioner provisioner;
+
+
+    @Override
+    public String name() {
+        return NAME;
+    }
+
+    @Override
+    public void initialize(ServiceExtensionContext context) {
+        // invoke once, so that the event registration definitely happens
+        createProvisioner(context);
+    }
+
+    @Provider
+    public AccountProvisioner createProvisioner(ServiceExtensionContext context) {
+        if (provisioner == null) {
+            var monitor = context.getMonitor().withPrefix("STS-Account");
+            if (stsClientStore != null) {
+                monitor.debug("This IdentityHub runtime contains an embedded SecureTokenService (STS) instance. ParticipantContexts and STS Clients will be synchronized.");
```

Comment:
Is this log message required? I don't think it provides any value, especially given the embedded STS extension initialization will be recorded.

Reply:
It doesn't hurt, since it's debug, and the presence of the STS extension alone doesn't necessarily mean that the synchronization happens.

Reply:
I generally agree with that, but this is not your average "entering method X" statement, but it highlights the availability of a significant feature, which would otherwise have to be inferred from the presence of one, and the absence of another log message, which is not convenient. 
I also don't think the message is cryptic.